src/zkevm_specs/tx.py

```diff
@@ -186,43 +213,49 @@ def assign(
         return cls(self_pub_key_hash, self_address, self_msg_hash_rlc, self_ecdsa_chip)
 
     def verify(self, keccak_table: KeccakTable, randomness: FQ, assert_msg: str):
-        is_enabled = FQ(1 - (self.msg_hash_rlc == 0))  # 1 - is_zero(self.msg_hash_rlc)
+        is_not_padding = FQ(1 - (self.address == 0))  # 1 - is_zero(self.address)
```

Comment:
Why the padding check is now using `address` instead `msg_hash_rlc`?

Reply:
Ah, good question!  Originally I intended to use `msg_hash == 0` to disable a transaction (and make it a padding one).  But for that I was assuming I would have an ECDSA Verify chip that would output 1 or 0 depending on whether the signature was correct.  But the current ECDSA Verify chip only supports correct signatures, and fails the constraints on invalid ones; so I decided to find a "dummy" input + witness values for padding transactions to pass the ECDSA verification.  And it turns out that `msg_hash = 0` is an [invalid value for a signature](https://crypto.stackexchange.com/questions/9379/is-there-any-weak-message-for-an-ecdsa-signature)!
So I had to change it to use `address == 0` to signal the padding transaction.  Since the address is the hash of the public key (which we verify via the keccak table), it's easy to deactivate the lookup when `address == 0` and use dummy values for the signature verification.

Reply:
Thanks for the detailed comment!

So, we need to remove this comment, isn't it?

https://github.com/appliedzkp/zkevm-specs/pull/187/files#diff-095d7a59798d973b0472d088a362302355f91c3acd241fc396758d705555c8d9R183

Reply:
> So, we need to remove this comment, isn't it?
> 
> https://github.com/appliedzkp/zkevm-specs/pull/187/files#diff-095d7a59798d973b0472d088a362302355f91c3acd241fc396758d705555c8d9R183

Yes!  Thanks for noticing this!  I've updated this in https://github.com/appliedzkp/zkevm-specs/pull/187/commits/8c37a3749a5788ab394e919ff97eafe351337d6c